app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java

```diff
@@ -152,34 +154,38 @@ private void startService(Context context)
 	}
 
 
-	private void showRestartVPN(final String message)
+	private void showRestartVPN(String message)
 	{
-		try
+		Intent intent = new Intent(context.getApplicationContext(), YonaActivity.class);
+		PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
+		if (pendingIntent == null)
```

Comment:
Actually the pending intent here is created by the VPN framework. So I thought of not taking chance. And there was also a [exception](https://fabric.io/yonadev/android/apps/nu.yona.app/issues/5c0982d2f8b88c2963d15d81/sessions/latest) registered in the Fabric in the same place.

Reply:
The documentation clearly states the return value cannot be null unless ``FLAG_NO_CREATE`` is given, so there is no need to test the return value for null.

The exception in Fabric is an NPE _inside_ ``PendingIntent.getActivity``.  The parameter ``intent``  was likely null. Check [the code for build 175](https://github.com/yonadev/yona-app-android/blob/aa2ff954f1be8f519104d71fcf5a31360c70d876/app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java):
```
			Intent intent = AppUtils.startVPN(context, true);
			PendingIntent pIntent = PendingIntent.getActivity(context, (int) System.currentTimeMillis() + 10000, intent, 0);
```

The implementation of ``AppUtils.startVPN`` has two code paths were it can return null. See the implementation [here](https://github.com/yonadev/yona-app-android/blob/aa2ff954f1be8f519104d71fcf5a31360c70d876/app/src/main/java/nu/yona/app/utils/AppUtils.java).